src/WCSimConstructRealisticPlacement.cc

```diff
@@ -0,0 +1,1168 @@
+    auto pmt20_dummy_logic = new G4LogicalVolume(pmt20_dummy_solid, pmt_mat, "pmt20");
+    G4VisAttributes* pmt20_dummy_colour = new G4VisAttributes(G4Colour(0.0,1.0,0.0));
+    pmt20_dummy_colour->SetForceLineSegmentsPerCircle(12);
+    pmt20_dummy_colour->SetForceAuxEdgeVisible(1);
+    pmt20_dummy_colour->SetForceWireframe(1);
+    pmt20_dummy_colour->SetLineWidth(3);
+    pmt20_dummy_logic->SetVisAttributes(pmt20_dummy_colour);
+
+    auto pmtmulti_dummy_solid = new G4Sphere("pmtMulti",0, 508*mm/2, 0, pi,0,pi);
+    auto pmtmulti_dummy_logic = new G4LogicalVolume(pmtmulti_dummy_solid, pmt_mat, "pmtMulti");
+    G4VisAttributes* pmtmulti_dummy_colour = new G4VisAttributes(G4Colour(1.0,0.0,0.0));
+    pmtmulti_dummy_colour->SetForceLineSegmentsPerCircle(12);
+    pmtmulti_dummy_colour->SetForceAuxEdgeVisible(1);
+    pmtmulti_dummy_colour->SetForceWireframe(1);
+    pmtmulti_dummy_colour->SetLineWidth(3);
+    pmtmulti_dummy_logic->SetVisAttributes(pmtmulti_dummy_colour);
+
+    auto pmtod_dummy_solid = new G4Sphere("pmtod",0, 200*mm/2, 0, pi,0,pi);
+    auto pmtod_dummy_logic = new G4LogicalVolume(pmtod_dummy_solid, pmt_mat, "pmtod");
+    G4VisAttributes* pmtod_dummy_colour = new G4VisAttributes(G4Colour(1.0,1.0,0.0));
+    pmtod_dummy_colour->SetForceLineSegmentsPerCircle(12);
+    pmtod_dummy_colour->SetForceAuxEdgeVisible(1);
+    pmtod_dummy_colour->SetForceWireframe(1);
+    pmtod_dummy_colour->SetLineWidth(3);
+    pmtod_dummy_logic->SetVisAttributes(pmtod_dummy_colour);
+
+
+    // *************************
+    // Barrel Assembly Placement
+    // *************************
+    // We have to make several nested assemblies to fill the full detector.
+    // Diagrams of each are kept with their respective components.
+    
+    // The hierarchy of assemblies are.
+    // - Barrel Side/Bottom Block
+    // - - PMT_only_row
+    // - - - PMT_only_two_cell
+    // - - - - pmt_offset_placement
+    // - - PMT_mPMT1_row
+    // - - - PMT_only_two_cell
+    // - - - - pmt_offset_placement
+    // - - - PMT_hybrid_three_cell
+    // - - - - pmt_offset_placement
+    // - - - - multipmt_offset_placement
+    // - - PMT_mPMT2_row
+    // - - - PMT_only_two_cell
+    // - - - - pmt_offset_placement
+    // - - - PMT_hybrid_three_cell
+    // - - - - pmt_offset_placement
+    // - - - - multipmt_offset_placement
+
+    // Six barrel side/bottom blocks are placed rotated around the barrel.
+    // Each PMT_only or PMT_mPMT row has frame cells (possible PMT holes) across.
+
+    // pmt_offset_placement/multipmt_offset_placement/pmtod_offset_placement
+    // ************************
+
+    // First make a single cell assembly, this will put the PMT flush against the wall
+    // Since assembly volumes just expand relative to their center point the means
+    // all nested volumes will also have the PMT againt the tyvek.
+    G4double InnerTyvekRadius = config.InnerDetectorOuterRadius;
```

Comment:
Unused variable compilation warning
```suggestion
    //G4double InnerTyvekRadius = config.InnerDetectorOuterRadius;
```